src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.WorkItem.cs

```diff
@@ -25,7 +26,21 @@ private partial class WorkCoordinator
 
                 // extra info
                 public readonly SyntaxPath ActiveMember;
-                public readonly ImmutableHashSet<IIncrementalAnalyzer> Analyzers;
+
+                /// <summary>
+                /// Non-empty if this work item is intended to be executed only for specific incremental analyzer(s).
+                /// Otherwise, the work item is applicable to all relevant incremental analyzers.
+                /// </summary>
+                public readonly ImmutableHashSet<IIncrementalAnalyzer> SpecificAnalyzers;
+
+                /// <summary>
+                /// Gets all the applicable analyzers to execute for this work item.
+                /// If this work item has any specific analyzer(s), then returns the intersection of <see cref="SpecificAnalyzers"/>
+                /// and the given <paramref name="allAnalyzers"/>.
+                /// Otherwise, returns <paramref name="allAnalyzers"/>.
+                /// </summary>
+                public IEnumerable<IIncrementalAnalyzer> GetApplicableAnalyzers(ImmutableArray<IIncrementalAnalyzer> allAnalyzers)
+                    => SpecificAnalyzers?.Count > 0 ? SpecificAnalyzers.Where(allAnalyzers.Contains) : allAnalyzers;
```

Comment:
Looks like `SpecificAnalyzers` can't be null? Consider turning on nullable-ref and make this obvious

Reply:
It is actually null for the default constructor for WorkItem, which I used in the last set of commits while using a dummy work item in ResetStatesAsync.